plumbum.py

```diff
@@ -89,16 +90,38 @@ def interpret_options(args=sys.argv[1:]):
     parser.add_argument("-r", "--region", help="AWS region", default=DEFAULT_REGION)
     parser.add_argument("-f", "--filter", help="filter to apply to AWS objects")
     parser.add_argument('--token', action='append', help='a key=value pair to use when populating templates')
-    parser.add_argument("namespace", type=str, help="AWS namespace")
     parser.add_argument("template", type=str, help="the template to interpret")
+    parser.add_argument("namespace", type=str, help="AWS namespace")
 
     args = parser.parse_args(args=args)
 
     # Support 'ec2' (human friendly) and 'AWS/EC2' (how CloudWatch natively calls these things)
-    namespace = args.namespace.rsplit('/', 2)[-1].lower()
+    if args.namespace is not None:  # Just making test pass, argparse will catch this missing.
+        namespace = args.namespace.rsplit('/', 2)[-1].lower()
+    else:
+        namespace = None
     return args.template, namespace, args.region, args.filter, args.token
 
 
+def list_billing(region, filter_by_kwargs):
+    """List available billing metrics"""
+    conn = boto.ec2.cloudwatch.connect_to_region(region)
+    metrics = conn.list_metrics(metric_name='EstimatedCharges')
+    # Filtering is based on metric Dimensions.  Only really valuable one is
+    # ServiceName.
+    if filter_by_kwargs:
+        filter_key = filter_by_kwargs.keys()[0]
+        filter_value = filter_by_kwargs.values()[0]
```

Comment:
Just to be clear, this will ignore all other pairs except the first one, right? And not even let the callers know?